dispatcher/client.go

```diff
@@ -35,7 +33,7 @@ type (
 		APIRequiredParameter RequiredParameter
 		APIOptionalParameter OptionalParameter
 
-		BuckupS3Option *BuckupS3Option
+		BuckupOption *BuckupOption
```

Comment:
1. BuckupOption will have some members. ( File name, Save Dir, etc...)
2. Struct is readable.

### Compare
```go
func (c *Client) SetBuckupFunction(do func(*os.File) error) {
	c.BuckupFunction = do
}
```

```go
...

client.SetBuckupFunction(s3.Upload)

client.SetBuckupOption(dispatcher.BuckupOption{
    Do: s3.Upload,
})

...
```